app/models/donation.rb

```diff
@@ -5,11 +5,20 @@ class Donation < ApplicationRecord
   belongs_to :user
 
   def self.donor_list
-    connection.select_all(
-      "SELECT who, SUM(amount) total, MAX(created_at) most_recent
-      FROM donations WHERE anonymous = 0 and reviewed = 1
-      GROUP BY who, email ORDER BY total DESC, most_recent DESC"
-    ).to_a
+    select_manager = arel_select_donor_list
+    # puts(select_manager.to_sql)
+    Donation.connection.select_all(select_manager.to_sql).to_a
+  end
+
+  # "SELECT who, SUM(amount) total, MAX(created_at) most_recent
+  # FROM donations WHERE anonymous = 0 and reviewed = 1
+  # GROUP BY who, email ORDER BY total DESC, most_recent DESC"
+  private_class_method def self.arel_select_donor_list
+    d = Donation.arel_table
+    d.where(d[:anonymous].eq(0).and(d[:reviewed].eq(1))).
+      project(d[:who], d[:amount].sum, d[:created_at].maximum).
+      group(d[:who], d[:email]).
+      order(d[:amount].sum.desc, d[:created_at].maximum.desc)
   end
```

Comment:
Nice, this looks good.

Reply:
I think it still makes sense to have mysql take care of the sum, etc. here.  Rewriting this in AR probably wouldn't buy us anything(?)

Reply:
By referencing the arel_tables in the pluck and order statements, we could do the sum in AR, the syntax would be almost the same. I think something like 
```suggestion
  Donation.where(anonymous: 0, reviewed: 1).group(:who, :email).
           order(d[:amount].sum.desc, d[:created_at].maximum.desc).
           pluck(:who, d[:amount].sum, d[:created_at].maximum)
  end
```

Reply:
1. I don't see how to do all of this in AR.
2. It would be nice to have a test assertion that tested the return value of donor list.